### PR TITLE
Check vm

### PR DESCRIPTION
**Pull request by govint**
Check for null VM name string before adding to list of vm names in generate_vm_list().

### CODE REVIEW

**Comment by govint** [comment]
Fixes #929

File: esx_service/cli/vmdkops_admin.py [hunk 1]
```diff
@@ -788,13 +788,15 @@ def generate_vm_list(vms_uuid):
...
     # vms_uuid is a list of vm_uuid
     # example: vms_uuid=["vm1_uuid", "vm2_uuid"]
-    # the return value is a string like this vm1,vm2
+    # the return value is a string like this vm1,vm2.
     res = ""
     for vm_uuid in vms_uuid:
         vm_name = vmdk_utils.get_vm_name_by_uuid(vm_uuid)
-        res = res + vm_name
+        if vm_name:
+            res = res + vm_name
```
Comment (pdhamdhere) [inline esx_service/cli/vmdkops_admin.py]: Add comment that VM may be deleted/not found.

File: esx_service/cli/vmdkops_admin.py [hunk 2]
```diff
@@ -788,13 +788,15 @@ def generate_vm_list(vms_uuid):
...
+        else:
+            res = res + vm_uuid
```
Comment (pdhamdhere) [inline esx_service/cli/vmdkops_admin.py]: This should never happen since tenant are ESX-local and only VMs on same ESX can be added to Tenant.

Comment (govint) [inline esx_service/cli/vmdkops_admin.py]: But the VM can migrate out of the host or be deleted. In the former case we still have the UUID from the KV but the VM name can't be resolved. Is there any way to keep the VMs of a tenant on the same host?

**Comment by govint** [comment]
Updated PR to ignore VMs in a tenant whose name can't be resolved.

File: esx_service/cli/vmdkops_admin.py [hunk 3]
```diff
@@ -788,13 +788,13 @@ def generate_vm_list(vms_uuid):
...
     # example: vms_uuid=["vm1_uuid", "vm2_uuid"]
-    # the return value is a string like this vm1,vm2
+    # the return value is a string like this vm1,vm2.
     res = ""
     for vm_uuid in vms_uuid:
         vm_name = vmdk_utils.get_vm_name_by_uuid(vm_uuid)
-        res = res + vm_name
-        res = res + ","
-
+        if vm_name:
```
Comment (pdhamdhere) [inline esx_service/cli/vmdkops_admin.py]: I would add a comment- "If VM is already deleted or migrated, vm_name could be NULL and will be skipped".

Comment (lipingxue) [inline esx_service/cli/vmdkops_admin.py]: I agree with Prashant.  Please add a comment here. Other LGTM.

**Comment by govint** [comment]
Added comment and below is the behavior with the fix. VM name is "1" and its ID is "2" below. First its added to the default tenant and shows up in "ls" and later unregistered and the command skips the VM 
when retried.

vmdkops_admin.py tenant ls
Uuid                                  Name      Description               Default_datastore  VM_list
------------------------------------  --------  ------------------------  -----------------  -------
cb15582e-7963-4d67-95ed-d1f5de21bb25  _DEFAULT  This is a default tenant                     1

vim-cmd vmsvc/unregister 2
(vim.fault.NotFound) {
   faultCause = (vmodl.MethodFault) null,
   faultMessage = <unset>
   msg = "Unable to find a VM corresponding to "2""
}
vmdkops_admin.py tenant ls
Uuid                                  Name      Description               Default_datastore  VM_list
------------------------------------  --------  ------------------------  -----------------  -------
cb15582e-7963-4d67-95ed-d1f5de21bb25  _DEFAULT  This is a default tenant

**Pull request merged**